### PR TITLE
Build from source with nginx.ng

### PR DESCRIPTION
**Pull request by grobinson-blockchain**
Build NGINX from source with nginx.ng state with support for passing
compile time flags necessary for installing modules such as more headers
by the openresty project.

### CODE REVIEW

File: nginx/ng/config.sls [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 {% from 'nginx/ng/map.jinja' import nginx, sls_block with context %}
 
+{% if nginx.install_from_source %}
+/var/log/nginx:
```
Comment (gravyboat) [inline nginx/ng/config.sls]: This sort of naming style is frowned upon, please use an identifier, then `- name: /var/log/nginx`.

File: nginx/ng/service.sls [hunk 2]
```diff
@@ -7,6 +7,12 @@
 
 include:
   - nginx.ng.install
+
+{% if nginx.install_from_source %}
+/lib/systemd/system/nginx.service:
```
Comment (gravyboat) [inline nginx/ng/service.sls]: Same issue with the naming here.

**Comment by gravyboat** [comment]
Should the source install live in the same state file? Or should it just be broken out into a unique install from source file? It adds a lot to the install state and might be cleaner? I'm not really passionate about either one, just a thought.

**Comment by grobinson-blockchain** [comment]
@gravyboat I'd be happy to move it into a separate state file. There was boilerplate for install from source in this sls with comments so I've just gone in an expanded on it.

**Comment by gravyboat** [comment]
Looks great, thanks for making these changes @grobinson-blockchain!

**Pull request merged**